c++/src/H5DataType.cpp

```diff
@@ -249,8 +249,19 @@ DataType::p_decode() const
         throw DataTypeIException("DataType::p_decode", "No encoded buffer");
     }
 
+    size_t tmp_buf_size = 0;
+
+    // Retrieve datatype size
+    if (H5Tencode(id, NULL, &tmp_buf_size) < 0) {
+        throw DataTypeIException("DataType::p_decode", "Unable to retrieve encoded buffer size");
+    }
+
+    if (tmp_buf_size == 0) {
+        throw DataTypeIException("DataType::p_decode", "Invalid encoded buffer size");
+    }
+
     // Call C function to decode the binary object description
-    hid_t encoded_dtype_id = H5Tdecode(encoded_buf);
+    hid_t encoded_dtype_id = H5Tdecode2(encoded_buf, tmp_buf_size);
 
```

Comment:
Thank you for addressing the issue, but I'm sorry, I made a mistake in the original comment.  If encoded_buf is not NULL then buf_size will already be the size of the encoded buffer.  You can safely use buf_size in H5Tdecode2 as you originally did.  I'm sorry again!

Reply:
Without having to call H5Tencode again to get the size, that is.  Checking buf_size is a good idea though.